Show total overdue patients count in tabs

msasikanth opened the pull request:
https://www.pivotaltracker.com/story/show/172931754

vinaysshenoy commented on app/src/main/java/org/simple/clinic/overdue/AppointmentRepository.kt:
```diff
@@ -129,6 +129,17 @@ class AppointmentRepository @Inject constructor(
+      since: LocalDate,
+      facility: Facility
+  ): Observable<List<Int>> {
+    return overdueDao.overdueAtFacilityCount(
```
What do you think of changing the repository method to return `Observable<Int>` instead and map the list by summing the values within the repository method? This way, the abstraction no longer leaks and we can change the query later without affecting consumers of this API.

msasikanth replied:
Does this makes sense?
```kt
    return overdueDao.overdueAtFacilityCount(
        facilityUuid = facility.uuid,
        scheduledBefore = since,
        scheduledAfter = since.minus(appointmentConfig.periodForIncludingOverdueAppointments)
    ).map { it.size }
```

vinaysshenoy replied:
Yup, looks good.

vinaysshenoy commented on app/src/main/java/org/simple/clinic/home/HomeScreenController.kt:
```diff
@@ -26,11 +32,49 @@ class HomeScreenController @Inject constructor(
+              facility = currentFacility
+          )
+        }
+        .replay()
+        .refCount()
```
This stream does not need to be shared since it has only one consumer.

msasikanth replied:
done

vinaysshenoy commented on app/src/main/java/org/simple/clinic/home/HomeScreenController.kt:
```diff
@@ -39,7 +83,8 @@ class HomeScreenController @Inject constructor(
               }
         }
-        .map { { ui: Ui -> ui.setFacility(it.name) } }
+        .replay()
+        .refCount()
```
The sharing of this stream should not be done here, but only in the call site where it is needed.

msasikanth replied:
Done

vinaysshenoy commented on app/src/main/java/org/simple/clinic/home/HomeScreen.kt:
```diff
@@ -99,10 +100,21 @@ class HomeScreen(context: Context, attrs: AttributeSet) : RelativeLayout(context
+    val overdueTab = homeTabLayout.getTabAt(overdueTabIndex)
+
+    overdueTab?.run {
+      val badge = orCreateBadge
```
Ugh, it took me a while to realise that this `orCreateBadge` is actually a method, `getOrCreateBadge` which Kotlin is treating as a getter method when it is not.

Can we use the complete method call instead?

vinaysshenoy replied:
I tried this, and added a `.apply {}` block to the badge.

```kotlin
overdueTab?.run {
  @Suppress("UsePropertyAccessSyntax")
  getOrCreateBadge().apply {
      isVisible = true
      maxCharacterCount = 3
      number = count 
  }
}
```

msasikanth replied:
Done

msasikanth commented:
> PR looks good, leaving it unapproved for now until we figure out what's going wrong with the gesture navigation.

Removed the migration to ViewPager 2 which was causing the issues. I have rebased the PR, so this will be the first commit 9acf5e0
